docs/source/executing-operations/defer-support.mdx

```diff
@@ -0,0 +1,170 @@
+
+## Using `@defer` with the router
+
+> To use `@defer` successfully with the Apollo Router, a client's GraphQL library must _also_ support the directive by handling "chunked" responses correctly.
+
+The Apollo Router supports deferring the following fields in your schema:
+
+* Fields that are nested anywhere within an [entity type](#entity-fields)
+* Root fields of the [`Query` type](#fields-of-query)
+
+In most supergraphs, this represents a large percentage of all fields in the schema. [See an example of unsupported fields.](#unsupported-field-example)
+
+### Entity fields
+
+The Apollo Router supports using `@defer` with fields and nested subfields of [entity types](/federation/entities/) in your federated supergraph. Entities are object types with at least one `@key`, which specifies the fields that can uniquely identify any given instance of that type.
+
+Here's an example subgraph schema that defines a `User` entity:
+
+```graphql title="Users subgraph"
+type User @key(fields: "id") {
+  id: ID!
+  name: Name!
+  friends: [User!]!
+}
+
+type Name {
+  firstName: String!
+  lastName: String!
+}
+
+type Query {
+  person(id: ID!): User
+}
+```
+
+Because the `User` type is an entity, a client query can apply `@defer` to any fragment that's nested somewhere within that type (at any depth).
+
+Here are two example queries:
+
+```graphql
+query QueryA($personId: ID!) {
+  person(id: $personId) {
+    id
+    # highlight-start
+    ... on User @defer {
+      name {
+        firstName
+        lastName
+      }
+    }
+    # highlight-end
+  }
+}
+
+query QueryB($personId: ID!) {
+  person(id: $personId) {
+    id
+    name {
+      firstName
+      # highlight-start
+      ... on Name @defer {
+        lastName
+      }
+      # highlight-end
+    }
+  }
+}
+```
+
+### Fields of `Query`
+
+The Apollo Router also supports using `@defer` with root fields of your supergraph's `Query` type:
+
+```graphql
+query PartiallyDeferredQuery {
+  products {
+    id
+  }
+  # highlight-start
+  ... on Query @defer {
+    users {
+      id
+    }
+  }
+  # highlight-end
+}
+```
+
+You _cannot_ use `@defer` with root fields of the `Mutation` type.
+
+### Unsupported field example
+
+Let's say our router's supergraph schema includes the following:
+
+```graphql
+type Book {
+  title: String!
+  author: String!
+}
+
+type Query {
+  books: [Book!]!
+}
+```
+
+The Apollo Router _cannot_ defer part of the following query as requested:
+
+<p style="margin-bottom: 0">⚠️</p>
+
+```graphql
+query DeferredAuthorQuery {
+  books {
+    title
+    # highlight-start
+    ... on Book @defer {
+      author # Can't be deferred
+    }
+    # highlight-end
+  }
+}
+```
+
+This is because `Book.author` is _not_ nested within an entity type. If the `Book` type were an entity, the router _would_ be able to defer this field.
```

Comment:
Fwiw, I'm not completely sure what the "nested" word is meant to refer to in this sentence. Are we're just trying to say that `Book.author` is not an entity field because `Book` is not an entity? If so, "nested" feels weird/unusual to me. If we mean something more involved by "nested", then I think this could use a bit more defining.

Reply:
My goal with "nested" is to emphasize that a field can be deferred as long as it appears anywhere within the hierarchy of an entity, regardless of how deep, so:

```graphql
a { # this is the entity
  b {
    c {
      d {
        ... @defer {
          e
        }
      }
    }
  }
}
```

Happy to take another stab at language here though if it feels ambiguous

Reply:
I mentioned it in my other comment on the updated content, but I'll repeat it here to be sure there isn't a further misunderstanding: what you describe above is actually *not* how it works. In that snippet above, it is not guaranteed that `e` can be deferred because `a` is an entity. What actually matters is if `d` is an entity (returns an entity type technically).

That's the subtlety we need to be careful with: nesting "within the query" is irrelevant. It is only nesting "within the deferred part" that has a role.

To hopefully clarify, I guess the algorithm to decide if a field is deferred is something like this:
1. start at the `@defer` and first look at the fields within it (and what happens before the defer is kind of irrelevant).
2. if the field is an entity field (it's parent type is an entity type) or root type field (it's parent type is a root type), then that field is going to be deferred alongside any of its sub-selection.
3. otherwise (the field is neither entity nor root type), that field is not deferred. If the field has some sub-selection, then re-run the steps on each fields of the sub-selection.

Not saying we should explain it that way btw. Just trying to remove any ambiguity.